webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -236,27 +245,10 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
     }
   }
 
-  private renderForm(formId) {
-    if (!formId) {
-      throw new Error('Unknown form');
-    }
-
+  private async renderForm(formContext: EnketoFormContext) {
     this.globalActions.setEnketoEditedStatus(false);
-    return this.dbService
-      .get()
-      .get(formId)
-      .then(form => {
-        const formInstanceData = this.getFormInstanceData();
-        const markFormEdited = this.markFormEdited.bind(this);
-        const resetFormError = this.resetFormError.bind(this);
-        return this.enketoService.renderContactForm(
-          '#contact-form',
-          form,
-          formInstanceData,
-          markFormEdited,
-          resetFormError
-        );
-      });
+
+    return this.enketoService.renderContactForm(formContext);
```

Comment:
We can take advantage of this function existing to use it as a place to define the form context:
```suggestion
  private async renderForm(formId, titleKey) {
    const formDoc = await this.dbService.get().get(formId);
    const instanceData = this.getFormInstanceData();
    const markFormEdited = this.markFormEdited.bind(this);
    const resetFormError = this.resetFormError.bind(this);
    const formContext: EnketoFormContext = {
      selector: '#contact-form',
      formDoc,
      instanceData,
      editedListener: markFormEdited,
      valuechangeListener: resetFormError,
      titleKey,
    };

    this.globalActions.setEnketoEditedStatus(false);

    return this.enketoService.renderContactForm(formContext);
  }
```